src/Mvc/Mvc.ViewFeatures/src/Rendering/HtmlHelperComponentExtensions.cs

```diff
@@ -62,6 +62,6 @@ public static class HtmlHelperComponentExtensions
 
         var viewContext = htmlHelper.ViewContext;
         var componentRenderer = viewContext.HttpContext.RequestServices.GetRequiredService<IComponentRenderer>();
-        return componentRenderer.RenderComponentAsync(viewContext, componentType, renderMode, parameters);
+        return await componentRenderer.RenderComponentAsync(viewContext, componentType, renderMode, parameters);
```

Comment:
why do you need the async?

Reply:
RenderComponentAsync is task-returning so we have to turn the `ValueTask` -> `Task`

Reply:
Could use
```suggestion
        return componentRenderer.RenderComponentAsync(viewContext, componentType, renderMode, parameters).AsTask();
```
instead to avoid the `async`-ceremony created by the compiler.